synapse/replication/tcp/protocol.py

```diff
@@ -254,13 +255,26 @@ async def handle_command(self, cmd: Command):
         # specific handling.
         cmd_func = getattr(self, "on_%s" % (cmd.NAME,), None)
         if cmd_func:
-            await cmd_func(cmd)
+            cmd_func(cmd)
             handled = True
 
         # Then call out to the handler.
         cmd_func = getattr(self.command_handler, "on_%s" % (cmd.NAME,), None)
         if cmd_func:
-            await cmd_func(self, cmd)
+            res = cmd_func(self, cmd)
+
+            # the handler might be a coroutine: fire it off as a background process
+            # if so.
+
+            if hasattr(res, "__await__"):
```

Comment:
I think I'd prefer to use `inspect.isawaitable` here

Reply:
done!